src/txkube/test/test_authentication.py

```diff
@@ -95,34 +145,36 @@ def _authorized_request(self, token, headers):
 
         self.patch(
             os, "environ", {
-                b"KUBERNETES_SERVICE_HOST": b"example.invalid.",
+                b"KUBERNETES_SERVICE_HOST": kubernetes_host,
                 b"KUBERNETES_SERVICE_PORT": b"443",
             },
         )
 
         agent = authenticate_with_serviceaccount(
             reactor, path=serviceaccount.path,
         )
-        agent.request(b"GET", b"http://example.invalid.", headers)
 
+        yield agent.request(b"GET", b"http://" + kubernetes_host, headers)
```

Comment:
I don't understand this change.  The agent isn't doing real network operations, is it?  Shouldn't it return an already-fired `Deferred`?  If so, `self.successResultOf` seems more appropriate - and there's no need to turn `_authorized_request` into a `Deferred`-returning API.

Reply:
This change was the easiest way for me to test the failure path by trying to resolve an invalid host against the FakeResolver, and the success path.
I didn't have good luck with successResultOf.